File: tests/fixtures/python-class-diagram/plugins/audio.py
"""Audio plugin with an inner class."""

from __future__ import annotations

from plugins.base import Plugin


class AudioPlugin(Plugin):
    _volume: float

    def __init__(self) -> None:
        self._volume = 0.8

    def activate(self) -> None:
        pass

    def deactivate(self) -> None:
        pass

    def set_volume(self, level: float) -> None:
        self._volume = level

    class Preset:
        """Inner class representing an audio preset."""

        name: str
        gain: float

        def __init__(self, name: str, gain: float) -> None:
            self.name = name
            self.gain = gain

File: tests/fixtures/go-class-diagram/plugin/audio.go
package plugin

// AudioPlugin plays audio content.
type AudioPlugin struct {
	volume float64
}

// Activate starts the audio subsystem.
func (a *AudioPlugin) Activate() {}

// Deactivate stops the audio subsystem.
func (a *AudioPlugin) Deactivate() {}

// SetVolume adjusts the playback volume.
func (a *AudioPlugin) SetVolume(level float64) {
	a.volume = level
}

// Preset is a nested type representing an audio preset.
type Preset struct {
	Name string
	Gain float64
}
